File: backend/src/main/java/com/example/Flicktionary/domain/series/dto/SeriesDetailResponse.java
package com.example.Flicktionary.domain.series.dto;

import com.example.Flicktionary.domain.director.dto.DirectorDto;
import com.example.Flicktionary.domain.genre.dto.GenreDto;
import com.example.Flicktionary.domain.series.entity.Series;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDate;
import java.util.Collections;
import java.util.List;
import java.util.Optional;

@AllArgsConstructor
@Builder
@Getter
public class SeriesDetailResponse {

    private Long id;

    private Long tmdbId;

    private String title;

    private String posterPath;

    private double averageRating;

    private int ratingCount;

    private Integer episode;

    private String plot;

    private String company;

    private String nation;

    private LocalDate releaseStartDate;

    private LocalDate releaseEndDate;

    private String status;

    private List<GenreDto> genres;

    private List<SeriesCastDto> casts;

    private DirectorDto director;

    public SeriesDetailResponse(Series series) {
        this.id = series.getId();
        this.tmdbId = series.getTmdbId();
        this.title = series.getTitle();
        this.posterPath = series.getPosterPath();
        this.averageRating = series.getAverageRating();
        this.ratingCount = series.getRatingCount();
        this.episode = series.getEpisodeNumber();
        this.plot = series.getOverview();
        this.company = series.getProductionCompany();
        this.nation = series.getProductionCountry();
        this.releaseStartDate = series.getReleaseStartDate();
        this.releaseEndDate = series.getReleaseEndDate();
        this.status = series.getStatus();

        this.genres = Optional.ofNullable(series.getGenres())
                .orElseGet(Collections::emptyList)
                .stream()
                .map(GenreDto::new)
                .toList();

        this.casts = Optional.ofNullable(series.getCasts())
                .orElseGet(Collections::emptyList)
                .stream()
                .map(SeriesCastDto::new)
                .toList();

        this.director = series.getDirector() != null ? new DirectorDto(series.getDirector()) : null;
    }
}

File: backend/src/main/kotlin/com/example/Flicktionary/domain/series/dto/SeriesDetailResponse.kt
package com.example.Flicktionary.domain.series.dto

import com.example.Flicktionary.domain.director.dto.DirectorDto
import com.example.Flicktionary.domain.genre.dto.GenreDto
import com.example.Flicktionary.domain.series.entity.Series
import lombok.AllArgsConstructor
import lombok.Builder
import lombok.Getter
import java.time.LocalDate

@AllArgsConstructor
@Builder
@Getter
class SeriesDetailResponse(
    val id: Long,
    val title: String,
    val posterPath: String?,
    val averageRating: Double,
    val ratingCount: Int,
    val episode: Int,
    val plot: String,
    val company: String,
    val nation: String,
    val releaseStartDate: LocalDate?,
    val releaseEndDate: LocalDate?,
    val status: String,
    val genres: List<GenreDto>,
    val casts: List<SeriesCastDto>,
    val director: DirectorDto?
) {
    constructor(series: Series) : this(
        series.id,
        series.title,
        series.posterPath,
        series.averageRating,
        series.ratingCount,
        series.episodeNumber,
        series.overview,
        series.productionCompany,
        series.productionCountry,
        series.releaseStartDate,
        series.releaseEndDate,
        series.status,
        genres = series.genres.map { GenreDto(it.id, it.name) },
        casts = series.casts.map { SeriesCastDto(it) },
        director = series.director?.let { DirectorDto(it) }
    )
}
